go/vt/vtgate/planbuilder/route.go

```diff
@@ -317,17 +317,26 @@ func (rb *route) PushSelect(expr *sqlparser.NonStarExpr, _ *route) (colsym *cols
 	colsym = newColsym(rb, rb.Symtab())
```

Comment:
PushSelect satisfies Builder (builder.go). The Join primitive that also supports this interface is the one that needs colnum (and possibly err). Any future primitives will also need to satisfy this signature.